stf.c

```diff
@@ -36,12 +38,16 @@ void setMC(float mc){
 }
 
 static void updateRealPolar(){
+  // we don't use Ideal Polar and degradation and ballast
+  // if we have the Real Polar from XCSoar
+  if (real_polar_valid) return;
+
   float loading_factor= (ideal_polar.w >0)? sqrt((ideal_polar.w + ballast) / ideal_polar.w) : 1;
   float deg_inv = (degradation>0)? 1.0/degradation : 1;
 
   polar.a= deg_inv * ideal_polar.a / loading_factor;
   polar.b= deg_inv * ideal_polar.b;
-  polar.c= deg_inv * ideal_polar.c / loading_factor;
+  polar.c= deg_inv * ideal_polar.c * loading_factor;
```

Comment:
Can you add some LI5 explanations for the math involved here? How did you come up with this change?

Reply:
See:
Streckensegelflug
Helmut Reichmann
ISBN 3-613-02479-9
Page 188 ff

Idaflieg.de various papers

In XCSoar:
GlidePolar::Update() in XCSoar/src/Engine/GlideSolvers/GlidePolar.cpp

Reply:
I suggest to add this reference as a comment, in case someone would want to change it.

Reply:
I'll add the comment in the next round after this PR got merged.
I believe there is an other mistake in the calculation of the load_factor.
It ought to be `loading_factor = sqrt(total_mass / reference_mass);`
I need to check the code again.

Reply:
As I said:
`sqrt((ideal_polar.w + ballast) / ideal_polar.w)`
is wrong.
then the equation must say:
`sqrt((dry_mass + ballast) / reference_mass)`
Whatever `ideal_polar.w` is it cannot be both and only in rare `cases dry_mass == reference_mass`